### DISCUSSION TITLE
Letting user decide which fields to show or hide

### DISCUSSION THREAD

**Discussion by Sebbito** [laravel/nova-issues#5277] [Ideas & Feature Requests]
## Problem
Hard coding the visibility of columns means you can only either have really bloated indexes or some fields missing missing from the index, for which you have to navigate to the detailview every time. This can become tedious if you are working with a large set of data.

## Solution
Let the user select which fields to show or hide for a given resource. Like a filter with boolean fields.

## Considerations
I have considered creating a filter and just listing all fields that the user should be able to manually show/hide but i have no idea how to do that and it would really bloat the filter section so i would rather not do it that way.

Is there maybe already a way to do this? I haven't seen this anywhere so i just assume this not a feature atm.

**Comment by davidhemphill** [comment]
I'm open to this feature as it would bring some value in specific contexts. Would there be an expectation that the values of the hidden fields wouldn't be selected from the database versus just hiding them from the user? If so, that's quite a bit bigger consideration.

**Reply by Sebbito** [reply]
The way i see it, all fields that are toggleable should be selected and loaded from the db the time you navigate to the index page.
If you toggle one fields visibility and the application then has to make a request to the db to load the fields values, it would take some time to load. Plus there is a lot of extra logic needed for a little optimization right?

I hope i understood your reply correctly ^^'

**Reply by pedro-santiago** [reply]
There could be an `optionalFields` method, there we could define how those fields look and feel. Those fields would appear in a dropdown.
